p2p/connectionpool/connectionpool.go

```diff
@@ -16,7 +16,7 @@ type dialResult struct {
 }
 
 type networker interface {
-	Dial(address string, remotePublicKey crypto.PublicKey) (net.Connection, error) // Connect to a remote node. Can send when no error.
+	Dial(address string, remotePublicKey cryptoBox.PublicKey) (net.Connection, error) // Connect to a remote node. Can send when no error.
```

Comment:
Please revert to the interface type defined in crypto.PulicKey. we need to be loose regarding the types we accept and strict about the types we return.

Reply:
@zalmen I guess only commitment identity should get a special name. all the rest (`SessionPublicKey`) are just public keys :/

Reply:
What I want to do is change it so that we have `p2p.PublicKey` or `p2pcrypto.PublicKey` and then it's clear we're talking about the public key used for p2p.

This should be distinct from any other public keys in the node that use a different scheme. In our case, NaCl defines the pubkey to be `Curve25519(ClampC(privkey), 9)` and the shared secret to be `Curve25519(ClampC(privkey), peerPubkey)`.

Reply:
done